### DISCUSSION TITLE
Mouse only emulation?

### DISCUSSION THREAD

**Discussion by jancheta** [tiny-pilot/tinypilot#492] [Q&A] [answered]
Hello, I was wondering if it was possible to make the tinypilot only forward mouse and look like a mouse device instead of keyboard/mouse combo?

**Comment by mtlynch** [comment, marked as answer]
It's not possible natively, but you could rewrite your `/opt/tinypilot-privileged/init-usb-gadget` file to make it act like a pure mouse rather than its current behavior (USB hub with mouse + keyboard inserted).

See this commit for an example of how I changed it to a keyboard-only device:

https://github.com/mtlynch/ansible-role-tinypilot/commit/dca65cf58e48c1ab399825aba783116e5e09b2dd

You'd have to take out the keyboard parts as opposed to the mouse parts.

You might also have to find a different value for `idProduct` because `0xa4ac` might be specifically for keyboard, but it's possible that it's for keyboard or mouse.

**Comment by croadfeldt** [comment]
This could be automated fairly easily, as well as adding in the mass storage item I just opened another discussion on. If there's desire, I could write up the automation for it while I am adding the mass storage item. Perhaps a select for enable / disable each component, along with the ability to upload / select disk images to mount? Would want some help with the UI portion, but I can do the rest.

**Reply by mtlynch** [reply]
Thanks!

The UI part is really what's challenging though. The user can achieve this today if they open up their [init-usb-gadget file](https://github.com/mtlynch/ansible-role-tinypilot/blob/0e65db74753df02eb77c181efd93aa91d8f4ef33/files/init-usb-gadget).

I'd want to hear more about the scenario that requires this before we start modifying the UI to support it.

**Reply by jancheta** [reply]
Im an embedded software engineer and been WFH bc covid...The product I am developing on now does not have X11 so I cannot use VNC as I have before. Thought this would be a slick way to get mouse input (paired with a webcam to see the integrated display) and control it while I am working from home. For some reason it seems like the presence of a keyboard messes with the mouse being picked up as well. I havent had much time to figure out exactly where the root issue is nor have I had the time to try your suggestions yet but I do plan to try them when I can squeeze it in.
